clang/include/clang/CConv/DeclRewriter.h

```diff
@@ -58,21 +58,20 @@ class DeclRewriter {
   // Rewrite a specific variable declaration using the replacement string in the
   // DAndReplace structure. Each of these functions is specialized to handling
   // one subclass of declarations.
-  void rewriteVarDecl(VarDeclReplacement *N, RSet &ToRewrite);
-  void rewriteFieldDecl(FieldDeclReplacement *N, RSet &ToRewrite);
   void rewriteParmVarDecl(ParmVarDeclReplacement *N);
+  template<typename DT>
+  void rewriteMultiDecl(DeclReplacementTempl<DT> *N, RSet &ToRewrite);
```

Comment:
Why do you mention the template name `DT` here but not in (say) `areDeclarationsOnSameLine`, below? Is it because you need to clarify you'll return the same type that you took in?

Reply:
In the other methods, I use the abstract base class as @jackastner suggested, so there's no template at all. However, here I need to call some methods that are on both `VarDecl` and `FieldDecl` but not on the `Decl` base class.